### ISSUE TITLE
How to used __call on Teal ?

### ISSUE THREAD

**Issue by Lurenlzm** [teal-language/tl#267]
example:

```
local record Point
end

function Point.new(): Point
    local self: Point = setmetatable({}, { __index = Point , __call = function() end })
    return self
end

local v = Point.new()
local c = v()  -- err

```

I used `--call` methods，but error .  How do I use it correctly?  @hishamhm

**Comment by hishamhm** [comment]
Compile-time knowledge about metamethods (including `__call`) was not implemented yet!

**Comment by hishamhm** [comment]
Metamethod support was merged in #299, it's now available in `master` ! :tada:

**Comment by winterwolf** [comment]
> Metamethod support was merged in #299, it's now available in `master` !

@hishamhm I have not checked it detailed and will let you know if find any problems, but thank you so much! I was really looking forward to it!

**Comment by lenscas** [comment]
wait, I forgot. What did I do again? :sweat_smile:
